From c6099bf8b04400e856747d51f3f880b509052935 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 3 Jan 2022 12:03:07 +0530
Subject: [PATCH] ath12k: fix qmi server arrive callback by correcting mhi
 channel configs

Channel execution environment mask expected by mhi driver differs from host
mhi config causes event stall after qmi init. Fix ee_mask value in mhi channel config.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mhi.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/mhi.c b/drivers/net/wireless/ath/ath12k/mhi.c
index 461c757..740985f 100644
--- a/drivers/net/wireless/ath/ath12k/mhi.c
+++ b/drivers/net/wireless/ath/ath12k/mhi.c
@@ -59,6 +59,9 @@ static const struct mhi_channel_config ath12k_mhi_channels_qcn9274[] = {
 		.offload_channel = false,
 		.doorbell_mode_switch = false,
 		.auto_queue = false,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
+		.auto_start = true,
+#endif
 	},
 	{
 		.num = 21,
@@ -73,6 +76,9 @@ static const struct mhi_channel_config ath12k_mhi_channels_qcn9274[] = {
 		.offload_channel = false,
 		.doorbell_mode_switch = false,
 		.auto_queue = true,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))		
+		.auto_start = true,
+#endif
 	},
 };
 
-- 
2.17.1

